File: DwindleDating/Backend/CustomSegue/CustomPopSegue.m
//
//  CustomPopSegue.m
//  SegueTest
//
//  Created by coeus on 12/06/2015.
//  Copyright (c) 2015 coeus. All rights reserved.
//

#import "CustomPopSegue.h"

@implementation CustomPopSegue

-(void)perform{
    UIViewController *sourceViewContreoller = [self sourceViewController];
    
    [sourceViewContreoller.navigationController popToRootViewControllerAnimated:TRUE];
}


@end

File: DwindleDating/Backend/CustomSegue/CustomPopSegue.swift
//
//  CustomPopSegue.swift
//  DwindleDating
//
//  Created by macbookpro on 12/06/2015.
//  Copyright (c) 2015 infinione. All rights reserved.
//

import UIKit

@objc(CustomPopSegue)

class CustomPopSegue: UIStoryboardSegue {

    
    func backViewController() -> UIViewController?{
        
        let sourceViewController = self.sourceViewController
        let navC = sourceViewController.navigationController!
        let noOfViewControllers = navC.viewControllers.count
        
        if (noOfViewControllers < 3){
            return nil
        }
        else{
            return navC.viewControllers[noOfViewControllers - 3]
        }
    }
    
    override func perform() {
        //
        let sourceViewController = self.sourceViewController
        let navC = sourceViewController.navigationController!
        navC.popToViewController(self.backViewController()!, animated: true)
    }
}